src/db/tables/onerep_scans.ts

```diff
@@ -72,35 +57,58 @@ async function setOnerepManualScan(
     onerep_profile_id: onerepProfileId,
     onerep_scan_id: onerepScanId,
     onerep_scan_reason: "manual",
+    // @ts-ignore knex.fn.now() results in it being set to a date,
+    // even if it's not typed as a JS date object:
     created_at: knex.fn.now(),
   });
 }
 
 async function setOnerepScanResults(
   onerepProfileId: number,
   onerepScanId: number,
-  onerepScanResults: object,
+  onerepScanResults: Array<ScanResult>,
   onerepScanReason: "manual" | "initial" | "monitoring"
 ) {
-  if (onerepScanReason === "manual") {
-    // Manual scans update an existing row.
-    await knex("onerep_scans")
-      .where("onerep_profile_id", onerepProfileId)
-      .andWhere("onerep_scan_id", onerepScanId)
-      .update({
-        onerep_scan_results: onerepScanResults,
-        updated_at: knex.fn.now(),
+  await knex.transaction(async (transaction) => {
+    if (onerepScanReason === "manual") {
+      // Manual scans update an existing scan, replacing the previous results:
+      await transaction("onerep_scan_results")
+        .delete()
+        .where("onerep_scan_id", onerepScanId);
+    } else {
+      // Initial and Monitoring scans always create a new scan:
+      await transaction("onerep_scans").insert({
+        onerep_profile_id: onerepProfileId,
+        onerep_scan_id: onerepScanId,
+        onerep_scan_reason: onerepScanReason,
+        // @ts-ignore knex.fn.now() results in it being set to a date,
+        // even if it's not typed as a JS date object:
+        created_at: knex.fn.now(),
       });
-  } else {
-    // Initial and Monitoring scans always create a new row.
-    await knex("onerep_scans").insert({
-      onerep_profile_id: onerepProfileId,
-      onerep_scan_id: onerepScanId,
-      onerep_scan_results: onerepScanResults,
-      onerep_scan_reason: onerepScanReason,
-      created_at: knex.fn.now(),
-    });
-  }
+    }
+
+    await transaction("onerep_scan_results").insert(
+      onerepScanResults.map((scanResult) => ({
+        onerep_scan_result_id: scanResult.id,
+        onerep_scan_id: scanResult.scan_id,
+        link: scanResult.link,
+        age:
+          typeof scanResult.age === "string"
+            ? Number.parseInt(scanResult.age, 10)
+            : undefined,
+        data_broker: scanResult.data_broker,
+        data_broker_id: scanResult.data_broker_id,
+        emails: JSON.stringify(scanResult.emails),
+        phones: JSON.stringify(scanResult.phones),
+        addresses: JSON.stringify(scanResult.addresses),
+        relatives: JSON.stringify(scanResult.relatives),
```

Comment:
No, I don't think it's important. IIRC this was just done because that's what the https://docs.onerep.com/#operation/getScanResult structure looks like, but I agree it's redundant in our case if there's nothing except the `data` key that we need.